$NetBSD: patch-configure.ac,v 1.31 2026/09/22 08:34:00 ryoon Exp $

* Add NetBSD/aarch64 support.
* Disable freetype2 and fontconfig from pkgsrc explicitly.
  to fix X11_TYPE=native build and internal pdfium's internal freetype build.
* Fix some static links.
* Find devel/apache-ant.
* Fix broken xinerama detection.
* Accept system/external box2d 3.1.1.
* Allow the system (pkgsrc) OpenSSL on Darwin.

--- configure.ac.orig
+++ configure.ac
@@ -5795,6 +5795,11 @@
         CPUNAME=X86_64
         RTL_ARCH=X86_64
         PLATFORMID=netbsd_x86_64
+        ;;
+    aarch64)
+        CPUNAME=AARCH64
+        RTL_ARCH=AARCH64
+        PLATFORMID=netbsd_aarch64
         ;;
     *)
         AC_MSG_ERROR([Unsupported host_cpu $host_cpu for host_os $host_os])
@@ -10455,9 +10460,9 @@
 dnl ===================================================================
 FREETYPE_CFLAGS_internal="${ISYSTEM}${WORKDIR}/UnpackedTarball/freetype/include"
 if test "x$ac_config_site_64bit_host" = xYES; then
-    FREETYPE_LIBS_internal="-L${WORKDIR}/UnpackedTarball/freetype/instdir/lib64 -lfreetype"
+    FREETYPE_LIBS_internal="${WORKDIR}/UnpackedTarball/freetype/instdir/lib64/libfreetype.a"
 else
-    FREETYPE_LIBS_internal="-L${WORKDIR}/UnpackedTarball/freetype/instdir/lib -lfreetype"
+    FREETYPE_LIBS_internal="${WORKDIR}/UnpackedTarball/freetype/instdir/lib/libfreetype.a"
 fi
 libo_CHECK_SYSTEM_MODULE([freetype],[FREETYPE],[freetype2 >= 9.9.3],,system,TRUE)
 if test "$_os" = WINNT -a "$ENABLE_HEADLESS" = TRUE; then
@@ -12002,15 +12007,7 @@
 if test "$enable_openssl" = "yes"; then
     AC_MSG_RESULT([no])
     ENABLE_OPENSSL=TRUE
-    if test "$_os" = Darwin ; then
-        # OpenSSL is deprecated when building for 10.7 or later.
-        #
-        # https://stackoverflow.com/questions/7406946/why-is-apple-deprecating-openssl-in-macos-10-7-lion
-        # https://stackoverflow.com/questions/7475914/libcrypto-deprecated-on-mac-os-x-10-7-lion
-
-        with_system_openssl=no
-        libo_CHECK_SYSTEM_MODULE([openssl],[OPENSSL],[openssl])
-    elif test "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o "$_os" = "OpenBSD" -o "$_os" = "DragonFly" \
+    if test "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o "$_os" = "OpenBSD" -o "$_os" = "DragonFly" \
             && test "$with_system_openssl" != "no"; then
         with_system_openssl=yes
         SYSTEM_OPENSSL=TRUE
@@ -12254,8 +12251,8 @@
     BOX2D_VERSION=`$PKG_CONFIG --modversion box2d 2>/dev/null`
     BOX2D_MAJOR=`echo $BOX2D_VERSION | cut -d"." -f1`
     BOX2D_MINOR=`echo $BOX2D_VERSION | cut -d"." -f2`
-    if test "$BOX2D_MAJOR" -eq "2"; then
-        if test "$BOX2D_MINOR" -eq "4"; then
+    if test "$BOX2D_MAJOR" -eq "3"; then
+        if test "$BOX2D_MINOR" -eq "1"; then
             _BOX2D_LIB=box2d
             AC_DEFINE(BOX2D_HEADER,<box2d/box2d.h>)
         else
@@ -13663,7 +13660,7 @@
     AC_MSG_RESULT([no (MPL only)])
 elif test "$enable_fuzzers" = "yes"; then
     AC_MSG_RESULT([no (oss-fuzz)])
-elif test \( \( "$_os" = "Linux" -o "$_os" = "Darwin" \) -a "$ENABLE_NSS" = TRUE \) -o "$_os" = "WINNT" ; then
+elif test \( \( "$_os" = "Linux" -o "$_os" = "Darwin" -o "$_os" = "NetBSD" \) -a "$ENABLE_NSS" = TRUE \) -o "$_os" = "WINNT" ; then
     AC_MSG_RESULT([yes])
     dnl ===================================================================
     dnl Check for system gpgme
@@ -14860,8 +14857,8 @@
         AC_CHECK_HEADER(X11/extensions/Xinerama.h, [],
             [AC_MSG_ERROR(Xinerama header not found.)], [])
         XEXTLIBS=`$PKG_CONFIG --variable=libs xext`
-        if test "x$XEXTLIB" = x; then
-           XEXTLIBS="-L$XLIB -L$XINERAMALIB -lXext"
+        if test "x$XEXTLIBS" = x; then
+           XEXTLIBS="-L$XINERAMALIB -lXext"
         fi
         XINERAMA_EXTRA_LIBS="$XEXTLIBS"
         if test "$_os" = "FreeBSD"; then
@@ -15241,8 +15238,8 @@
                     if test -f /usr/share/ant-core/lib/ant.jar; then
                         ANT_LIB=/usr/share/ant-core/lib
                     else
-                        if test -f $ANT_HOME/lib/ant/ant.jar; then
-                            ANT_LIB="$ANT_HOME/lib/ant"
+                        if test -f $ANT_HOME/lib/java/ant/ant.jar; then
+                            ANT_LIB="$ANT_HOME/lib/java/ant"
                         else
                             if test -f /usr/share/lib/ant/ant.jar; then
                                 ANT_LIB=/usr/share/lib/ant
